### PR TITLE
dooit: 1.0.1 -> 2.0.1

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes
Version bump and new dependency added. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 255047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>dooit</li>
    <li>dooit.dist</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 255047` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>dooit</li>
    <li>dooit.dist</li>
  </ul>
</details>

File: pkgs/tools/misc/dooit/default.nix [hunk 1]
```diff
@@ -5,34 +5,27 @@
...
     pyperclip
+    python-dateutil
     pyyaml
-    dateparser
+    textual
     tzlocal
-    appdirs
   ];
 
   # No tests available
```
Comment (Artturin) [inline pkgs/tools/misc/dooit/default.nix:31]: The  package is missing a `pythonImportsCheck` test

Comment (Artturin) [reply pkgs/tools/misc/dooit/default.nix:31]: oh this is not a module

Comment (Artturin) [reply pkgs/tools/misc/dooit/default.nix:31]: oh this is not a module nvm that.

Comment (Artturin) [reply pkgs/tools/misc/dooit/default.nix:31]: Please add a https://nixos.org/manual/nixpkgs/stable/#tester-testVersion

Comment (khaneliman) [reply pkgs/tools/misc/dooit/default.nix:33]: Pushed a test, we'll see what ofborg sees. Locally it still says 2.0.0 from the doot --version

Comment (Artturin) [reply pkgs/tools/misc/dooit/default.nix:33]: The dev forgot to bump the version https://github.com/kraanzu/dooit/compare/v2.0.1...main and only did it after the release, use `substituteInPlace` in `postPatch` to fix it

Comment (khaneliman) [reply pkgs/tools/misc/dooit/default.nix:33]: Ah, lame. I'll do it when I get back tonight. Thanks for catching that

**Comment by khaneliman** [comment]
@Artturin should be good now

**Comment by Artturin** [comment]
@wesleyjrz  ping

**Comment by wesleyjrz** [comment]
Result of nixpkgs-review pr 255047 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

<details>
<summary>2 packages built:</summary>

- dooit
- dooit.dist

</details>

**Review by wesleyjrz — APPROVED** [review]
Result of nixpkgs-review pr 255047 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

<details>
<summary>2 packages built:</summary>

- dooit
- dooit.dist

</details>

**Pull request merged**